docs/contributing.md

```diff
@@ -179,3 +179,11 @@ To debug code generation the flag &#x2013;typescript.debugger can be used
 - Make sure to re-build autorest.typescript after any changes you want to test
 
       npm run build
+
+### How to update dependencies in generated package.json files?
+
+Our generated SDKs depend on various packages. Usually, these packages are set to some specific versions in the generator. During the course of time, we need to update these dependencies. Sometimes, this might be due to the requirements of supporting new features. Or, this might be the result of compliance violations. There are CI Jobs which are setup to monitor the compliance violations in the SDK Generator repositories. For the JS SDK Repository, the CI Job could be found [here](https://dev.azure.com/azure-sdk/internal/_componentGovernance/101977/alert/5809940?typeId=7613604). Now, these violations should be resolved permanently.
+
+Such dependencies are added to the `package.json` file in the [`packageFileGenerator.ts`](https://github.com/Azure/autorest.typescript/blob/main/src/generators/static/packageFileGenerator.ts). This file has 2 methods `restLevelPackage` & `regularAutorestPackage`. Determine which package is causing the violation and make the appropriate changes. For example, let us say `@opentelemetry/api` package of version `0.10.2` is causing a compliance violation. The solution is to upgrade the version to `1.0.3`. (Usually, such recommendations are found in the compliance issue itself) So, change the version and create a PR. (Please note that these changes would require `npm run build && npm run generate-swaggers && npm run smoke-tests` before creating the PR) Once the PR is merged & released, any and all SDKs generated with the new version will not have the compliance issue.
+
+Now, there is a scenario where there will be a compliance violation and there is no such dependency entry in the SDK generator. But, the SDKs in the JS Repository will be having that dependency. For example, there was a violation reported for the `nyc` package. The issue was resolved through this [PR](https://github.com/Azure/azure-sdk-for-js/pull/19248). But, there is no dependency entry for `nyc` in the [`packageFileGenerator.ts`](https://github.com/Azure/autorest.typescript/blob/main/src/generators/static/packageFileGenerator.ts). It means that entry has been made manually and not part of the SDK generator. In such cases, you do not have to do any code changes in the generator. (It is sufficient to do the changes manually in the JS Repo) But, Please raise the absence of dependency as an issue in the repository. This will help the Generator owners to consider adding that dependency to the [`packageFileGenerator.ts`](https://github.com/Azure/autorest.typescript/blob/main/src/generators/static/packageFileGenerator.ts).
```

Comment:
When we need to upgrade a dependency to use a new feature, there will be many other changes in the generated code as well. Anyone doing that will be knowing how to update the version in generated package.json file as well. So, we don't need to mention that particularly here. I would also stay away from all talks of compliance violations for the same reasons I mentioned in https://github.com/Azure/autorest.typescript/pull/1264#discussion_r773401058




```suggestion
Our generated SDKs take dependency on various packages which you can see in the generated package.json files. These will need to be upgraded from time to time to stay on the latest major version so that we get bug fixes automatically due to semver.

Such dependencies are added to the `package.json` file in the [`packageFileGenerator.ts`](https://github.com/Azure/autorest.typescript/blob/main/src/generators/static/packageFileGenerator.ts). Find the dependency you are trying to update in either of the 2 methods `restLevelPackage` & `regularAutorestPackage` here and make the change. Run `npm run build && npm run generate-swaggers && npm run smoke-tests` before creating the PR.
```